core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -835,7 +835,7 @@ case class Stop(id: stop.Stop, sourceInfo: SourceInfo, clock: Arg, ret: Int) ext
   "This API should never have been public, for Module port reflection, use DataMirror.modulePorts",
   "Chisel 3.5"
 )
-case class Port(id: Data, dir: SpecifiedDirection)
+case class Port(id: Data, dir: SpecifiedDirection, sourceInfo: SourceInfo = UnlocatableSourceInfo)
```

Comment:
```suggestion
case class Port(id: Data, dir: SpecifiedDirection, sourceInfo: SourceInfo)
```
I would rather not make this have a default because I want us to be cognizant in our internals about whether or not we have a source locator to pass rather than being able to accidentally forget because of the default.

Reply:
will fix on next push